core/commands/ipns.go

```diff
@@ -107,16 +109,24 @@ Resolve the value of a dnslink:
 		}
 
 		recursive, _, _ := req.Option("recursive").Bool()
-		depth := 1
-		if recursive {
-			depth = namesys.DefaultDepthLimit
+		rc, rcok, _ := req.Option("dht-record-count").Int()
+		dhtt, dhttok, _ := req.Option("dht-timeout").Int()
+		opts := namesys.DefaultResolveOpts()
+		if !recursive {
+			opts.Depth = 1
+		}
+		if rcok {
+			opts.DhtRecordCount = uint(rc)
```

Comment:
Can get rid of this cast once https://github.com/ipfs/go-ipfs-cmdkit/pull/14 is approved